### PR TITLE
add function to list dependencies of a template

### PR DESCRIPTION
**Pull request by dmfenton**
Get a list of the unique values a template depends upon.

e.g.
```js
const { listDependencies } = require('adlib')
const template = 'Injuries: {{CRASHID}}<br />Fatalities: {{ISREPORTONSCENE}} Date: {{Date:toIso}}'
const deps = new Set(listDependencies(template));
assert.deepEqual(deps, new Set('CRASHID', 'ISREPORTONSCENE'));
```

### CODE REVIEW

**Comment by dmfenton** [comment]
apparently in node you cannot maintain the api of
```
adlib()
adlib.function
```

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "adlib",
   "version": "2.2.5",
   "description": "Templating for deep JSON object graphs",
-  "main": "dist/adlib.umd.js",
+  "main": "dist/adlib.js",
```
[the remark this answers is not in the document]
Comment (dmfenton) [reply package.json]: adlib.umd.js is not actually generated but adlib.js is and its a UMD.

File: test/adlib.tape.js [hunk 2]
```diff
@@ -79,6 +79,27 @@ test('Adlib::Strings:: allow info-window template to pass through', (t) => {
...
 
+test('Adlib::listDependencies:: parse dependecies from a string', (t) => {
+  const template = 'Injuries: {{CRASHID}}<br />Fatalities: {{ISREPORTONSCENE}} Foo: {{CRASHID}}'
+  const deps = new Set(adlib.listDependencies(template));
+  t.deepEqual(deps, new Set('CRASHID', 'ISREPORTONSCENE'));
+  t.end();
+})
+
+test('Adlib::listDependencies:: parse dependecies from a string with transforms', (t) => {
+  const template = 'Injuries: {{CRASHID}}:toIso<br />Fatalities: {{ISREPORTONSCENE}} Foo: {{CRASHID}}'
```
[the remark this answers is not in the document]
Comment (dmfenton) [reply test/adlib.tape.js]: fixed

File: test/adlib.tape.js [hunk 3]
```diff
@@ -79,6 +79,27 @@ test('Adlib::Strings:: allow info-window template to pass through', (t) => {
   t.end();
 })
 
+test('Adlib::listDependencies:: parse dependecies from a string', (t) => {
```
[the remark this answers is not in the document]
Comment (dmfenton) [reply test/adlib.tape.js]: added

**Comment by dmfenton** [comment]
@dbouwman I've updated this according to your feedback

**Comment by dmfenton** [comment]
The ember addon is already looking for dist/adlib.js
https://github.com/ArcGIS/ember-arcgis-adlib-service/blob/master/index.js#L26

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbouwman) [reply package.json]: ah - well then 👍

**Comment by dmfenton** [comment]
It works as is with 
adlib()
adlib.listDependencies()

If you don’t mind the slightly awkward api then I think we should merge as is and avoid a breaking change.

**Comment by dbouwman** [comment]
@dmfenton works for me - lmk if you don't have privs to publish this package

**Pull request merged**